skyportal/plot.py

```diff
@@ -2041,11 +2061,11 @@ def make_spectrum_layout(obj, spectra, user, device, width, smoothing, smooth_nu
         normfac = np.nanmedian(np.abs(s.fluxes))
         normfac = normfac if normfac != 0.0 else 1e-20
         altdata = json.dumps(s.altdata) if s.altdata is not None else ""
-        annotations = (
-            AnnotationOnSpectrum.query_records_accessible_by(user)
-            .filter(AnnotationOnSpectrum.spectrum_id == s.id)
-            .all()
-        )
+        annotations = session.scalars(
+            AnnotationOnSpectrum.select(session.user_or_token).where(
```

Comment:
This is nice. You should be aware that it only works with a handler's `self.Session()` and will not work with a `DBSession()`. I don't know if that is going to be a problem or not. Probably not.

Reply:
Makes sense, for now we pass a handler session (which should hopefully always be the case)